src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -321,18 +321,22 @@ internal BoundExpression BindToNaturalType(BoundExpression expression, BindingDi
                             Debug.Assert(op.NoCommonTypeError != 0);
                             type = CreateErrorType();
                             hasErrors = true;
-                            object trueArg = op.Consequence.Display;
-                            object falseArg = op.Alternative.Display;
-                            if (op.NoCommonTypeError == ErrorCode.ERR_InvalidQM && trueArg is Symbol trueSymbol && falseArg is Symbol falseSymbol)
+
+                            if (!op.HasAnyErrors)
```

Comment:
@AlekseyTs On one hand, as I already stated, error reported in this branch, spans over the whole expression, so it makes sense to disable it if there are any other "more local" errors in the expression. On the other hand, logically this error is only related to true and false branches so it also makes sence to suppress it only if there are errors specifically in them. For instance, consider existing test with expression `1 ? null : null`. It has error on `1` since it is not convertible to boolean, but also there is no conversion betwen nulls in both branches. How do you think, should we report error on the whole expression about no conversion between nulls in branches in such case?